### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by onedisha**
THIS IS MY FIRST PR , YAY ! (LOVEEE HOGGY <3)

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -24,8 +22,7 @@ taskList.addEventListener("click", function (event) {
         const checkbox = event.target;
         const taskText = checkbox.nextElementSibling;
         const listItem = checkbox.parentNode;
-        const changeTo= checkbox.checked;
-        editDb(listItem.id, changeTo);
+        const changeTo = checkbox.checked;
         if (checkbox.checked) {
```
Comment (yudeeeth) [inline script.js:27]: add comments for this part of the code to expain what you are doing

File: script.js [hunk 2]
```diff
@@ -143,6 +110,7 @@ function editDb(tempid, changeTo){
         body: JSON.stringify({id: tempid, checked: changeTo})
     })
     .then(res => res.json())
+    .then(res => getDb())
     .catch(err => console.log("no data to edit!"));
```
Comment (yudeeeth) [inline script.js]: Throw an alert here instead

File: script.js [hunk 3]
```diff
@@ -131,6 +97,7 @@ function deleteFromDb(tempid){
         body: JSON.stringify({id: tempid})
     })
     .then(res => res.json())
+    .then(res => getDb())
     .catch(err => console.log("no data to delete!"));
```
Comment (yudeeeth) [inline script.js]: alert here too

File: script.js [hunk 4]
```diff
@@ -118,8 +84,8 @@ function addToDb(obj){
         body: JSON.stringify(obj)
     })
     .then(res => res.json())
-    .then(res => data.push(res))
-    .catch(err => console.log("no data!"));
+    .then(res => getDb())
+    .catch(err => console.log("push unsuccessful"));
```
Comment (yudeeeth) [inline script.js]: Add an alert here please

File: script.js [hunk 5]
```diff
@@ -24,8 +22,7 @@ taskList.addEventListener("click", function (event) {
         const checkbox = event.target;
         const taskText = checkbox.nextElementSibling;
         const listItem = checkbox.parentNode;
-        const changeTo= checkbox.checked;
-        editDb(listItem.id, changeTo);
+        const changeTo = checkbox.checked;
         if (checkbox.checked) {
             taskText.classList.add("checked");
             for (let i=0; i<data.length; i++){
```
Comment (yudeeeth) [inline script.js]: convert this to data.filter

File: express-http-server/app.js [hunk 6]
```diff
@@ -1,43 +1,29 @@
...
-
-// app.post('/data', (req, res) => {
-//     const obj = req.body.data;
-//     backendData=obj;
-//     res.json("Data added successfully");
-//     console.log(backendData);
-// })
-
 app.get('/all', async (req, res) => {
     const ret = await tododata.find({}).toArray();
```
Comment (yudeeeth) [inline express-http-server/app.js:28]: why are you using .toArray here please explain

File: express-http-server/app.js [hunk 7]
```diff
@@ -1,43 +1,29 @@
...
+const { ObjectId } = require('mongodb');
 
 // server config
 const app = express();
 app.use(express.json());
 app.use(cors());
 const port = 4000;
 
-const { MongoClient } = require("mongodb");
 const uri = "mongodb://127.0.0.1:27017";
```
Comment (yudeeeth) [inline express-http-server/app.js:13]: add a .env file for this as this is supposed to be a secret

Comment (yudeeeth) [reply express-http-server/app.js:13]: the uri i mean

**Review by yudeeeth — CHANGES_REQUESTED** [review]
Please check these

File: script.js (hunk 1 above)
Comment (onedisha) [reply script.js:27]: OKII

**Pull request merged**